app/models/user.rb

```diff
@@ -0,0 +1,30 @@
+class User < ActiveRecord::Base
+  has_secure_password
+  has_many :questions
+  has_many :answers
+  has_many :comments
+  has_many :votes
+
+  def reputation
+    question_rep = self.questions.map { |q| q.score }.inject(:+)
```

Comment:
We can do these in SQL - e.g. `answer_rep = self.answers.sum(:score)`

You could also do the whole thing in one query by writing sql.